OpenAFSLibrary/keywords/path.py

```diff
@@ -234,18 +236,39 @@ def get_fid(self, path):
         fid = "%d.%d.%d" % (int(volume), int(vnode), int(unique))
         return fid
 
+    def _rilookup(self, fid):
+        """ 
+        This is not a keyword method. It uses RPC to query RIDB on file 
+        server.
+        This needs py_openafs installed.
+        Simplest way is to add the dir to $PYTHONPATH environment variable.
+        """
+        rx.rx_Init(0, 0)
+        conn = rx.rx_NewConnection('127.0.0.1', rx.RXAFS_port, rx.RXAFS_service_id)
```

Comment:
This hard codes the address as localhost. This is the fileserver host, correct? What if we are running on a cluster? How do we know which host to contact?

Reply:
Since we have the FID, that will tell us the volume. Whichever file server is serving that volume should be the one.

Reply:
We'll for the test we have, we create the volume on a given server, so we know the fileserver at that point, as we assume no one moved it, and it is a read-write volume so we know it exists on only one file server.  In general tho, a fid could be for a file in a read-only volume, so could be on any one of the fileservers that have copies of the read-only volume.

Reply:
vikram's project assumes RW vols at this point, so assuming a single server is okay. I can think of a few ways to get the fileserver addr:

- lookup the fid's vol in the vldb, and use the first fileserver
- lookup the fid's vol in the vldb, and return a list of results (one per fileserver)
- take a fileserver as an arg to the robotest keyword or whatever it's called

I'm not sure which matches the style or conventions of robotest stuff

Reply:
My suggestion is to add an optional argument (that is, a keyword style argument) to the the new get_name_by_fid() keyword.   The current test has the fileserver hostname, since it was used in the test setup to create the test volume containing the file under test.  Later, If we want, we can add a vl lookup in the keyword implementation if the fileserver hostname was not passed.